src/main/java/org/aion/wallet/ui/components/SendController.java

```diff
@@ -63,26 +70,37 @@ public void onSendAionClicked() {
             dto = mapFormData();
         } catch (ValidationException e) {
             log.error(e.getMessage(), e);
+            setErrorStatus(e.getMessage());
             return;
         }
-        txStatusLabel.setText("Sending transaction...");
+        setNormalStatus(PENDING_MESSAGE);
 
         final Task<String> sendTransactionTask = getApiTask(this::sendTransaction, dto);
 
         runApiTask(
                 sendTransactionTask,
                 evt -> handleTransactionFinished(sendTransactionTask.getValue()),
-                getErrorEvent(t -> txStatusLabel.setText(t.getMessage()), sendTransactionTask),
+                getErrorEvent(t -> Optional.ofNullable(t.getCause()).ifPresent(cause -> setErrorStatus(cause.getMessage())), sendTransactionTask),
                 getEmptyEvent()
         );
     }
 
     private void handleTransactionFinished(final String txHash) {
-        log.info("Transaction finished: " + txHash);
-        txStatusLabel.setText("Transaction Finished");
+        log.info("%s: %s", SUCCESS_MESSAGE, txHash);
+        setNormalStatus(SUCCESS_MESSAGE);
         EventPublisher.fireOperationFinished();
     }
 
+    private void setErrorStatus(final String message) {
```

Comment:
wouldn't it be better to have only one method and handle the different options with case ?